### PR TITLE
 Define and test pod find endpoint.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
Implement the pod controllers find ability.

JIRA issues: MARATHON-7876

### CODE REVIEW

File: src/test/scala/mesosphere/marathon/api/akkahttp/v2/ResponseMatchers.scala [hunk 1]
```diff
@@ -40,11 +40,11 @@ trait ResponseMatchers { this: UnitTest =>
...
     */
-  def definedNetworkname(name: String) = new HavePropertyMatcher[JsValue, Option[String]] {
+  def definedNetworkName(name: String) = new HavePropertyMatcher[JsValue, Option[String]] {
     override def apply(actual: JsValue) = {
-      val maybeNetworkname = (actual \ "networks" \ 0 \ "name").asOpt[String]
-      val matches = maybeNetworkname.contains(name)
-      HavePropertyMatchResult(matches, "networkname", Some(name), maybeNetworkname)
+      val maybeNetworkName = (actual \ "networks" \ 0 \ "name").asOpt[String]
+      val matches = maybeNetworkName.contains(name)
+      HavePropertyMatchResult(matches, "networkName", Some(name), maybeNetworkName)
```
Comment (jeschkies) [inline src/test/scala/mesosphere/marathon/api/akkahttp/v2/ResponseMatchers.scala]: @alenkacz, thanks for your persistence.

File: src/main/scala/mesosphere/marathon/api/akkahttp/Rejections.scala [hunk 2]
```diff
@@ -17,14 +17,19 @@ object Rejections {
   case class EntityNotFound(message: Message = defaultEntityNotFoundMessage) extends Rejection
   object EntityNotFound {
     def noApp(id: PathId, version: Option[Timestamp] = None): EntityNotFound = {
-      EntityNotFound(Message(s"App '$id' does not exist" + version.fold("")(v => s" in version $v")))
+      val readableVersion = version.fold("")(v => s" in version $v")
+      EntityNotFound(Message(s"App '$id' does not exist $readableVersion"))
     }
     def noGroup(id: PathId, version: Option[Timestamp] = None): EntityNotFound = {
-      EntityNotFound(Message(s"Group '$id' does not exist" + version.fold("")(v => s" in version $v")))
+      val readableVersion = version.fold("")(v => s" in version $v")
```
Comment (meln1k) [inline src/main/scala/mesosphere/marathon/api/akkahttp/Rejections.scala]: What do you think if we factor this to a common method?

File: src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala [hunk 3]
```diff
@@ -55,7 +59,12 @@ class PodsControllerTest extends UnitTest with ScalatestRouteTest with RouteBeha
       val request = Post(Uri./.withQuery(Query("force" -> "false")))
         .withEntity(entity)
         .withHeaders(`Remote-Address`(RemoteAddress(InetAddress.getByName("192.168.3.12"))))
+
+      val podDefinition = PodDefinition(id = PathId("mypod"))
+      f.podManager.find(any).returns(Some(podDefinition))
```
Comment (meln1k) [inline src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala]: Can we explicitly set the id here? For example: `f.podManager.find(equals(id)).returns(Some(podDefinition))`

File: src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala [hunk 4]
```diff
@@ -388,6 +397,31 @@ class PodsControllerTest extends UnitTest with ScalatestRouteTest with RouteBeha
         jsonResponse should have(executorResources(cpus = 100.0, mem = 100.0, disk = 10.0))
       }
     }
+
+    "respond with a pod for a lookup" in {
+      val f = Fixture()
+      val controller = f.controller()
+
+      val podDefinition = PodDefinition(id = PathId("mypod"))
+      f.podManager.find(any).returns(Some(podDefinition))
```
Comment (meln1k) [inline src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala]: Let's use an ID here :)

File: src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala [hunk 5]
```diff
@@ -388,6 +397,31 @@ class PodsControllerTest extends UnitTest with ScalatestRouteTest with RouteBeha
...
+        val jsonResponse = Json.parse(responseAs[String])
+        jsonResponse should have(podId("mypod"))
+      }
+    }
+
+    "reject a lookup a specific pod that pod does not exist" in {
+      val f = Fixture()
+      val controller = f.controller()
+
+      f.podManager.find(any).returns(Option.empty[PodDefinition])
```
Comment (meln1k) [inline src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala]: And here.

File: src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala (hunk 3 above)
Comment (jeschkies) [reply src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala]: Good point.

Comment (jeschkies) [reply src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala]: ✅

File: src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala (hunk 4 above)
Comment (jeschkies) [reply src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala]: ✅

File: src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala (hunk 3 above)
Comment (jeschkies) [reply src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala]: Here we do not care what parameters are passed. We test the authorization for a pod.

File: src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala (hunk 5 above)
Comment (jeschkies) [reply src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala]: ✅

File: src/main/scala/mesosphere/marathon/api/akkahttp/Rejections.scala (hunk 2 above)
Comment (jeschkies) [reply src/main/scala/mesosphere/marathon/api/akkahttp/Rejections.scala]: Yeah, let's do it.

**Pull request merged**